electrum/gui/qt/main_window.py

```diff
@@ -1964,12 +1964,15 @@ def on_pr(self, request: 'paymentrequest.PaymentRequest'):
         else:
             self.payment_request_error_signal.emit()
 
-    def parse_lightning_invoice(self, invoice):
+    def set_ln_invoice(self, invoice: LNInvoice):
         """Parse ln invoice, and prepare the send tab for it."""
         try:
             lnaddr = lndecode(invoice, expected_hrp=constants.net.SEGWIT_HRP)
-        except Exception as e:
-            raise LnDecodeException(e) from e
+        except (LnDecodeException, LnAddressError) as e:
+            self.show_error(_("Error parsing Lightning invoice") + f":\n{e}")
+            return
```

Comment:
This looks suspect.
Note that `lndecode` has a bit unfortunate interface: it can raise any Exception. This is why the old code is catching Exception and raising instead a more specific exception (LnDecodeException) so that callers can distinguish what happened.
`lndecode` itself is not raising `LnDecodeException`. We could change that potentially, but that is simply not how it works atm.
My point is, with your suggested change, this block would "leak" e.g. ValueErrors and so on.